patches/react-native-web+0.19.9+002+fix-mvcp.patch

```diff
@@ -0,0 +1,687 @@
+     };
+@@ -633,12 +638,10 @@ class VirtualizedList extends StateSafePureComponent {
+     };
+     this._getFrameMetrics = (index, props) => {
+       var data = props.data,
+-        getItem = props.getItem,
+         getItemCount = props.getItemCount,
+         getItemLayout = props.getItemLayout;
+       invariant(index >= 0 && index < getItemCount(data), 'Tried to get frame for out of range index ' + index);
+-      var item = getItem(data, index);
+-      var frame = this._frames[this._keyExtractor(item, index, props)];
++      var frame = this._frames[VirtualizedList._getItemKey(props, index)];
+       if (!frame || frame.index !== index) {
+         if (getItemLayout) {
+           /* $FlowFixMe[prop-missing] (>=0.63.0 site=react_native_fb) This comment
+@@ -662,7 +665,7 @@ class VirtualizedList extends StateSafePureComponent {
+ 
+       // The last cell we rendered may be at a new index. Bail if we don't know
+       // where it is.
+-      if (focusedCellIndex >= itemCount || this._keyExtractor(props.getItem(props.data, focusedCellIndex), focusedCellIndex, props) !== this._lastFocusedCellKey) {
++      if (focusedCellIndex >= itemCount || VirtualizedList._getItemKey(props, focusedCellIndex) !== this._lastFocusedCellKey) {
+         return [];
+       }
+       var first = focusedCellIndex;
+@@ -702,9 +705,15 @@ class VirtualizedList extends StateSafePureComponent {
+       }
+     }
+     var initialRenderRegion = VirtualizedList._initialRenderRegion(_props);
++    var minIndexForVisible = (_this$props$maintainV = (_this$props$maintainV2 = this.props.maintainVisibleContentPosition) == null ? void 0 : _this$props$maintainV2.minIndexForVisible) !== null && _this$props$maintainV !== void 0 ? _this$props$maintainV : 0;
+     this.state = {
+       cellsAroundViewport: initialRenderRegion,
+-      renderMask: VirtualizedList._createRenderMask(_props, initialRenderRegion)
++      renderMask: VirtualizedList._createRenderMask(_props, initialRenderRegion),
++      firstVisibleItemKey: this.props.getItemCount(this.props.data) > minIndexForVisible ? VirtualizedList._getItemKey(this.props, minIndexForVisible) : null,
++      // When we have a non-zero initialScrollIndex, we will receive a
++      // scroll event later so this will prevent the window from updating
++      // until we get a valid offset.
++      pendingScrollUpdateCount: this.props.initialScrollIndex != null && this.props.initialScrollIndex > 0 ? 1 : 0
+     };
+ 
+     // REACT-NATIVE-WEB patch to preserve during future RN merges: Support inverted wheel scroller.
+@@ -715,7 +724,7 @@ class VirtualizedList extends StateSafePureComponent {
+       var clientLength = this.props.horizontal ? ev.target.clientWidth : ev.target.clientHeight;
+       var isEventTargetScrollable = scrollLength > clientLength;
+       var delta = this.props.horizontal ? ev.deltaX || ev.wheelDeltaX : ev.deltaY || ev.wheelDeltaY;
+-      var leftoverDelta = delta;
++      var leftoverDelta = delta * 0.5;
```

Comment:
@getusha what's this change for? It caused https://github.com/Expensify/App/issues/31402

Reply:
root cause here https://github.com/Expensify/App/issues/31402#issuecomment-1814988967

Reply:
@rushatgabhane @situchan this is not something new, i just applied a patch from https://github.com/Expensify/react-native-web/pull/16/files

and was not removed until now https://github.com/Expensify/react-native-web/blob/master/packages/react-native-web/src/vendor/react-native/VirtualizedList/index.js#L815, i am pretty sure it helped with scroll issues we had.